src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -305,7 +339,15 @@ class ActivityFeed extends React.Component<Props, State> {
      */
     updateTaskAssignment = (taskId: string, taskAssignmentId: string, status: string): void => {
         const updateTaskAssignment = this.props.onTaskAssignmentUpdate || noop;
-        updateTaskAssignment(taskId, taskAssignmentId, status);
+        const { feedItems } = this.state;
+        const task = feedItems.find((item) => !!(item.id === taskId));
+        if (!task) {
+            return;
+        }
+
+        updateTaskAssignment(taskId, taskAssignmentId, status, (updatedAssignment) =>
+            this.updateTaskAssignmentSuccessCallback(task, updatedAssignment)
```

Comment:
When you add the failure case, please update the item with the error object like we do in other failure cases.